plugins/modules/pnp_workflow_manager.py

```diff
@@ -1259,6 +1189,151 @@ def verify_diff_deleted(self, config):
+
+        Parameters:
+        - self (object): An instance of the class containing the method.
+        - pnp_params (dict): Device infomation to add in the inventry
+        Returns:
+        The method returns an instance of the class with updated attributes:
+        - self.device_response: Gets the device_id and stores it
+
+        Example:
+        passing device details and getting pnp device details response
+        """
+        try:
+            device_add_response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function="add_device",
+                params=pnp_params,
+                op_modifies=True,
+            )
+            if device_add_response:
+                return device_add_response
+
+        except Exception as e:
+            msg = "Unable to add the PNP device {0}.".format(str(pnp_params))
+            self.log(msg + str(e), "WARNING")
+            self.module.fail_json(msg=msg)
+
+    def pnp_device_count(self, pnp_params):
+        """
+        Get the PNP device count from the Cisco Catalyst Center.
+
+        Parameters:
+            - self (object): An instance of the class containing the method.
+            - pnp_params (dict): Device infomation to get the pnp device fields
+        Returns:
+            The method returns an instance of the class with count attributes:
+            - prov_dev_response: Show the count of the pnp devices
+
+        Example:
+            passing device param and getting pnp device count response
+        """
+        try:
+            prov_dev_response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function='get_device_count',
+                op_modifies=True,
+                params=pnp_params,
+            )
+            if prov_dev_response:
+                return prov_dev_response
+
+        except Exception as e:
+            msg = "Unable to get the PNP device count for {0}.".format(str(pnp_params))
+            self.log(msg + str(e), "WARNING")
+            self.module.fail_json(msg=msg)
+
+    def claim_device_site(self, claim_params):
+        """
+        Claim the PNP device from the Cisco Catalyst Center.
+
+        Parameters:
+            - self (object): An instance of the class containing the method.
+            - claim_params (dict): Device infomation to get the pnp device fields
+        Returns:
+            The method returns an instance of the class with count attributes:
+            - claim_response: Show the count of the pnp devices
+
+        Example:
+            passing device claim param and getting pnp claim response
+        """
+        try:
+            claim_response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function='claim_a_device_to_a_site',
+                op_modifies=True,
+                params=claim_params,
+            )
+            if claim_response:
```

Comment:
self.log("Successfully claimed device to site: {}".format(claim_response), "INFO")
            return claim_response

        # If the response is empty, log a warning
        msg = "No response received when trying to claim the device to site with parameters: {}".format(claim_params)
        self.log(msg, "WARNING")
        self.module.fail_json(msg=msg)

Reply:
updated the same